src/centralstore/incremental/NodeManager.cpp

```diff
@@ -0,0 +1,254 @@
+        node_manager_logger.warn("Relation/Edge already exist for source = " + std::string(source.id) +
+                                 " destination = " + std::string(destination.id));
+    }
+    return 0;
+}
+
+NodeBlock *NodeManager::addNode(std::string nodeId) {
+    unsigned int assignedNodeIndex;
+    if (this->nodeIndex.find(nodeId) == this->nodeIndex.end()) {
+        node_manager_logger.debug("NodeId not found in index for node ID " + nodeId);
+        NodeBlock *sourceBlk = new NodeBlock(nodeId, this->nextNodeIndex * NodeBlock::BLOCK_SIZE);
+        this->nodeIndex.insert({nodeId, this->nextNodeIndex});
+        assignedNodeIndex = this->nextNodeIndex;
+        this->nextNodeIndex++;
+        sourceBlk->save();
+        return sourceBlk;
+    } else {
+        node_manager_logger.debug("NodeId found in index for node ID " + nodeId);
+        return this->get(nodeId);
+    }
+}
+
+std::pair<NodeBlock, NodeBlock> NodeManager::addEdge(std::pair<std::string, std::string> edge) {
+    NodeBlock *sourceNode = this->addNode(edge.first);
+    NodeBlock *destNode = this->addNode(edge.second);
+    unsigned int relationAddr = this->addRelation(*sourceNode, *destNode);
+
+    node_manager_logger.debug("DEBUG: Source DB block address " + std::to_string(sourceNode->addr) +
+                              " Destination DB block address " + std::to_string(destNode->addr));
+    std::pair<NodeBlock, NodeBlock> returnVal = {*sourceNode, *destNode};
+    delete sourceNode; // to prevent memory leaks
+    delete destNode;
+    return returnVal;
+}
+
+int NodeManager::dbSize(std::string path) {
+    /*
+        The structure stat contains at least the following members:
+        st_dev     ID of device containing file
+        st_ino     file serial number
+        st_mode    mode of file (see below)
+        st_nlink   number of links to the file
+        st_uid     user ID of file
+        st_gid     group ID of file
+        st_rdev    device ID (if file is character or block special)
+        st_size    file size in bytes (if file is a regular file)
+        st_atime   time of last access
+        st_mtime   time of last data modification
+        st_ctime   time of last status change
+        st_blksize a filesystem-specific preferred I/O block size for
+                        this object.  In some filesystem types, this may
+                        vary from file to file
+        st_blocks  number of blocks allocated for this object
+    */
+    struct stat result;
+    if (stat(path.c_str(), &result) == 0) {
+        node_manager_logger.debug("Size of the " + path + " is " + std::to_string(result.st_size));
+    } else {
+        node_manager_logger.error("Error while reading file stats in " + path);
+        return -1;
+    }
+
+    return result.st_size;
+}
+
+/**
+ * @Deprecated use NodeBlock.get() instead
+ **/
+NodeBlock *NodeManager::get(std::string nodeId) {
+    NodeBlock *nodeBlockPointer = NULL;
+    int debug = std::stoi(nodeId);
```

Comment:
This variable debug is not used in this function.

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/ed31ba0af33e9629173808960eef3cfe90c5dbe9

Reply:
Done